components/concept/storage/src/main/java/mozilla/components/concept/storage/CreditCardsStorage.kt

```diff
@@ -0,0 +1,202 @@
+
+    /**
+     * Deletes the credit card with the given [guid].
+     *
+     * @return True if the deletion did anything, false otherwise.
+     */
+    suspend fun deleteCreditCard(guid: String): Boolean
+
+    /**
+     * Marks the credit card with the given [guid] as `in-use`.
+     *
+     * @param guid Unique identifier for the desired credit card.
+     */
+    suspend fun touchCreditCard(guid: String)
+
+    /**
+     * Inserts the provided address into the database, and returns
+     * the newly added [Address].
+     *
+     * @param addressFields A [NewAddressFields] record to add.
+     * @return [Address] for the added address.
+     */
+    suspend fun addAddress(addressFields: NewAddressFields): Address
+
+    /**
+     * Retrieves the address from the underlying storage layer by its unique identifier.
+     *
+     * @param guid Unique identifier for the desired address.
+     * @return [Address] record.
+     */
+    suspend fun getAddress(guid: String): Address
+
+    /**
+     * Retrieves a list of all the addresses.
+     *
+     * @return A list of all [Address].
+     */
+    suspend fun getAllAddresses(): List<Address>
+
+    /**
+     * Updates the fields in the provided address.
+     *
+     * @param address An [Address] record to update.
+     */
+    suspend fun updateAddress(address: Address)
+
+    /**
+     * Delete the address with the given [guid].
+     *
+     * @return True if the deletion did anything, false otherwise.
+     */
+    suspend fun deleteAddress(guid: String): Boolean
+
+    /**
+     * Marks the address with the given [guid] as `in-use`.
+     *
+     * @param guid Unique identifier for the desired address.
+     */
+    suspend fun touchAddress(guid: String)
+}
+
+/**
+ * Information about a credit card.
+ *
+ * @property guid The unique identifier for this credit card.
+ * @property billingName The credit card billing name.
+ * @property cardNumber The credit card number.
+ * @property expiryMonth The credit card expiry month.
+ * @property expiryYear The credit card expiry year.
+ * @property cardType The credit card network ID.
+ */
+data class CreditCard(
+    val guid: String,
+    val billingName: String,
+    val cardNumber: String,
+    val expiryMonth: Int,
+    val expiryYear: Int,
+    val cardType: String
+)
+
+/**
+ * Information about a new credit card.
+ *
+ * @property billingName The credit card billing name.
+ * @property cardNumber The credit card number.
+ * @property expiryMonth The credit card expiry month.
+ * @property expiryYear The credit card expiry year.
+ * @property cardType The credit card network ID.
+ */
+data class NewCreditCardFields(
+    val billingName: String,
+    val cardNumber: String,
+    val expiryMonth: Int,
+    val expiryYear: Int,
+    val cardType: String
+)
+
+/**
+ * Information about a address.
+ *
+ * @property guid The unique identifier for this address.
+ * @property givenName First name.
```

Comment:
I would prefer `firstName` here

Reply:
I think it would be preferable to be consistent with the naming across the different layers like AS https://github.com/mozilla/application-services/blob/main/components/autofill/src/autofill.udl#L56 and desktop https://searchfox.org/mozilla-central/source/browser/extensions/formautofill/FormAutofillUtils.jsm#20 for anyone reading into multiple implementations of this.